tensorflow/python/lib/io/file_io.py

```diff
@@ -778,6 +779,26 @@ def list_directory_v2(path):
       for filename in _pywrap_file_io.GetChildren(compat.path_to_bytes(path))
   ]
 
+@tf_export("io.gfile.join")
```

Comment:
> We should add tests for this new API endpoint in file_io_test.py

(from internal review)

Reply:
That's a good idea.
I added some tests in `file_io_test.py`, very similar to what was already in `ram_file_system_test.py ` but also testing `gcs://` and that the behavior for native filesystems.